src/Microsoft.Extensions.DependencyInjection.Specification.Tests/DependencyInjectionSpecificationTests.cs

```diff
@@ -0,0 +1,546 @@
+            Assert.NotNull(service1.TransientService.FakeService);
+
+            Assert.Equal(42, service2.TransientService.Value);
+            Assert.NotNull(service2.TransientService.FakeService);
+
+            Assert.NotNull(service1.ScopedService.FakeService);
+
+            // Verify scoping works
+            Assert.NotSame(service1.TransientService, service2.TransientService);
+            Assert.Same(service1.ScopedService, service2.ScopedService);
+        }
+
+        [Fact]
+        public void LastServiceReplacesPreviousServices()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddTransient<IFakeMultipleService, FakeOneMultipleService>();
+            collection.AddTransient<IFakeMultipleService, FakeTwoMultipleService>();
+            var container = CreateContainer(collection);
+
+            // Act
+            var service = container.GetService<IFakeMultipleService>();
+
+            // Assert
+            Assert.IsType<FakeTwoMultipleService>(service);
+        }
+
+        [Fact]
+        public void SingletonServiceCanBeResolved()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddSingleton<IFakeSingletonService, FakeService>();
+            var container = CreateContainer(collection);
+
+            // Act
+            var service1 = container.GetService<IFakeSingletonService>();
+            var service2 = container.GetService<IFakeSingletonService>();
+
+            // Assert
+            Assert.NotNull(service1);
+            Assert.Same(service1, service2);
+        }
+
+        [Fact]
+        public void ServiceContainerRegistersServiceScopeFactory()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            var container = CreateContainer(collection);
+
+            // Act
+            var scopeFactory = container.GetService<IServiceScopeFactory>();
+
+            // Assert
+            Assert.NotNull(scopeFactory);
+        }
+
+        [Fact]
+        public void ScopedServiceCanBeResolved()
+        {
+            // Arrange
+            var collection = new ServiceCollection();
+            collection.AddScoped<IFakeScopedService, FakeService>();
+            var container = CreateContainer(collection);
+
+            // Act
+            var scopeFactory = container.GetRequiredService<IServiceScopeFactory>();
+            using (var scope = scopeFactory.CreateScope())
+            {
+                var containerScopedService = container.GetService<IFakeScopedService>();
+                var scopedService1 = scope.ServiceProvider.GetService<IFakeScopedService>();
+                var scopedService2 = scope.ServiceProvider.GetService<IFakeScopedService>();
+
+                // Assert
+                Assert.NotEqual(containerScopedService, scopedService1);
+                Assert.Equal(scopedService1, scopedService2);
```

Comment:
`NotSame` and `Same`? Just for consistency with the other tests. I also think it conveys the intent clearer.